golang/btcspv/types_test.go

```diff
@@ -55,38 +57,26 @@ func TestTypes(t *testing.T) {
 	spvProof := new(SPVProof)
 	err = json.Unmarshal([]byte(typesSuite.Fixtures.Valid[0]), &spvProof)
 	logIfErr(err)
-	typesSuite.Proof = *spvProof
+	typesSuite.ValidProof = *spvProof
 
 	for i := 0; i < len(typesSuite.Fixtures.BadHeaders); i++ {
-		bitcoinHeader := new(BitcoinHeader)
-		err = json.Unmarshal([]byte(typesSuite.Fixtures.BadHeaders[i]), &bitcoinHeader)
+		invalidHeader := new(InvalidHeadersCases)
+		err = json.Unmarshal([]byte(typesSuite.Fixtures.BadHeaders[i]), &invalidHeader)
 		logIfErr(err)
-		appended := append(typesSuite.BadHeaders, *bitcoinHeader)
-		typesSuite.BadHeaders = appended
-	}
 
-	for i := 0; i < len(typesSuite.Fixtures.BadHeaderErrors); i++ {
-		headerErr := new(Err)
-		err = json.Unmarshal([]byte(typesSuite.Fixtures.BadHeaderErrors[i]), &headerErr)
-		logIfErr(err)
-		appended := append(typesSuite.BadHeaderErrors, *headerErr)
-		typesSuite.BadHeaderErrors = appended
+		invalidHeader.Error = strings.Replace(invalidHeader.Error, "\u00a0", " ", -1)
```

Comment:
In the invalid header cases, the error strings come back with regex, i.e. `"MerkleRootLE\u00a0is\u00a0not\u00a0the\u00a0LE\u00a0version\u00a0of\u00a0MerkleRoot"`.  This messes with tests so I had to filter them out.  I don't understand why we have this problem here, but not with the invalid proof cases.